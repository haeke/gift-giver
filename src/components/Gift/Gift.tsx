import React, { Component } from "react";
import { Form, FormGroup, FormControl, Button } from "react-bootstrap";

interface State {
  person: string;
  present: string;
}

class Gift extends Component<{}, State> {
  state = {
    person: "",
    present: ""
  };
  render() {
    return (
      <div>
        <Form>
          <FormControl
            className="input-person"
            onChange={event => this.setState({ person: event.target.value })}
          />
          <FormControl
            className="input-present"
            onChange={event => this.setState({ present: event.target.value })}
          />
        </Form>
      </div>
    );
  }
}

export default Gift;
