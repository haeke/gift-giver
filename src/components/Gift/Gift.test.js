import React from "react";
// Shallow renders the root node and returns a shallow wrapper around it. It must be a single-node wrapper.
//
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Gift from "./Gift";

// required to make enzyme work
configure({ adapter: new Adapter() });

describe("Gift", () => {
  const mockRemove = jest.fn();
  const id = 1;
  const props = { gift: { id }, removeGift: mockRemove };
  const gift = shallow(<Gift {...props} />);

  it("expects Gift to work", () => {
    expect(gift).toMatchSnapshot();
  });

  it("expects the `state` to contain an object with person and present properties", () => {
    expect(gift.state()).toEqual({ person: "", present: "" });
  });

  describe("when typing into the the person input", () => {
    const person = "Uncle";
    beforeEach(() => {
      // desribes testing the person input such that it receives a value of Uncle
      gift
        .find(".input-person")
        .simulate("change", { target: { value: person } });
    });

    it("updates the person object in `state`", () => {
      expect(gift.state().person).toEqual(person);
    });
  });

  describe("when typing into the present input", () => {
    const present = "Calculator";
    beforeEach(() => {
      // test the present input such that it received a value of Calculator
      gift
        .find(".input-present")
        .simulate("change", { target: { value: present } });
    });

    it("updates the present object in `state`", () => {
      expect(gift.state().present).toEqual(present);
    });
  });

  describe("when clicking on the remove gift button", () => {
    beforeEach(() => {
      gift.find(".btn-remove").simulate("click");

      it("called the removeGift function", () => {
        expect(mockRemove).toHaveBeenCalledWith(id);
      });
    });
  });
});
