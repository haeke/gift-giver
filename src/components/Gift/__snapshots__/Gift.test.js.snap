// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift expects Gift to work 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gift />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ForwardRef
        as="form"
        inline={false}
      >
        <ForwardRef
          as="input"
          className="input-person"
          onChange={[Function]}
        />
        <ForwardRef
          as="input"
          className="input-present"
          onChange={[Function]}
        />
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "form",
        "children": Array [
          <ForwardRef
            as="input"
            className="input-person"
            onChange={[Function]}
          />,
          <ForwardRef
            as="input"
            className="input-present"
            onChange={[Function]}
          />,
        ],
        "inline": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "input",
            "className": "input-person",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Feedback": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
                "type": "valid",
              },
              "displayName": "Feedback",
              "propTypes": Object {
                "as": [Function],
                "type": [Function],
              },
              "render": [Function],
            },
            "defaultProps": Object {
              "as": "input",
            },
            "displayName": "FormControl",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "input",
            "className": "input-present",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Feedback": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
                "type": "valid",
              },
              "displayName": "Feedback",
              "propTypes": Object {
                "as": [Function],
                "type": [Function],
              },
              "render": [Function],
            },
            "defaultProps": Object {
              "as": "input",
            },
            "displayName": "FormControl",
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Check": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Input": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "type": "checkbox",
            },
            "displayName": "FormCheckInput",
            "render": [Function],
          },
          "Label": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "type": "checkbox",
            },
            "displayName": "FormCheckLabel",
            "render": [Function],
          },
          "defaultProps": Object {
            "disabled": false,
            "inline": false,
            "isInvalid": false,
            "isValid": false,
            "title": "",
            "type": "checkbox",
          },
          "displayName": "FormCheck",
          "render": [Function],
        },
        "Control": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Feedback": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
              "type": "valid",
            },
            "displayName": "Feedback",
            "propTypes": Object {
              "as": [Function],
              "type": [Function],
            },
            "render": [Function],
          },
          "defaultProps": Object {
            "as": "input",
          },
          "displayName": "FormControl",
          "render": [Function],
        },
        "Group": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "FormGroup",
          "render": [Function],
        },
        "Label": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "column": false,
            "srOnly": false,
          },
          "displayName": "FormLabel",
          "render": [Function],
        },
        "Row": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "FormRow",
          "render": [Function],
        },
        "Text": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "small",
          },
          "displayName": "FormText",
          "render": [Function],
        },
        "defaultProps": Object {
          "as": "form",
          "inline": false,
        },
        "displayName": "Form",
        "render": [Function],
      },
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ForwardRef
          as="form"
          inline={false}
        >
          <ForwardRef
            as="input"
            className="input-person"
            onChange={[Function]}
          />
          <ForwardRef
            as="input"
            className="input-present"
            onChange={[Function]}
          />
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "form",
          "children": Array [
            <ForwardRef
              as="input"
              className="input-person"
              onChange={[Function]}
            />,
            <ForwardRef
              as="input"
              className="input-present"
              onChange={[Function]}
            />,
          ],
          "inline": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "input",
              "className": "input-person",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Feedback": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                  "type": "valid",
                },
                "displayName": "Feedback",
                "propTypes": Object {
                  "as": [Function],
                  "type": [Function],
                },
                "render": [Function],
              },
              "defaultProps": Object {
                "as": "input",
              },
              "displayName": "FormControl",
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "input",
              "className": "input-present",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Feedback": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                  "type": "valid",
                },
                "displayName": "Feedback",
                "propTypes": Object {
                  "as": [Function],
                  "type": [Function],
                },
                "render": [Function],
              },
              "defaultProps": Object {
                "as": "input",
              },
              "displayName": "FormControl",
              "render": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Check": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Input": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "type": "checkbox",
              },
              "displayName": "FormCheckInput",
              "render": [Function],
            },
            "Label": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "type": "checkbox",
              },
              "displayName": "FormCheckLabel",
              "render": [Function],
            },
            "defaultProps": Object {
              "disabled": false,
              "inline": false,
              "isInvalid": false,
              "isValid": false,
              "title": "",
              "type": "checkbox",
            },
            "displayName": "FormCheck",
            "render": [Function],
          },
          "Control": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Feedback": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
                "type": "valid",
              },
              "displayName": "Feedback",
              "propTypes": Object {
                "as": [Function],
                "type": [Function],
              },
              "render": [Function],
            },
            "defaultProps": Object {
              "as": "input",
            },
            "displayName": "FormControl",
            "render": [Function],
          },
          "Group": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "FormGroup",
            "render": [Function],
          },
          "Label": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "column": false,
              "srOnly": false,
            },
            "displayName": "FormLabel",
            "render": [Function],
          },
          "Row": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "FormRow",
            "render": [Function],
          },
          "Text": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "small",
            },
            "displayName": "FormText",
            "render": [Function],
          },
          "defaultProps": Object {
            "as": "form",
            "inline": false,
          },
          "displayName": "Form",
          "render": [Function],
        },
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
